pyiron_base/project/generic.py

```diff
@@ -1700,23 +1700,19 @@ def _remove_jobs_helper(self, recursive=False, progress=True):
         """
         if not isinstance(recursive, bool):
             raise ValueError("recursive must be a boolean")
-        if not self.db.view_mode:
-            job_id_lst = self.get_job_ids(recursive=recursive)
-            if progress and len(job_id_lst) > 0:
-                job_id_lst = tqdm(job_id_lst)
-            for job_id in job_id_lst:
-                if job_id not in self.get_job_ids(recursive=recursive):
-                    continue
```

Comment:
If I understand this pull request correctly, then the primary change is removing this database query, correct? Maybe we should split this pull request in two parts. One containing just the removal of this part and the other the refactoring. Having both mixed makes it a bit hard to read.

Reply:
To clarify these changes a bit more I created two separate pull requests https://github.com/pyiron/pyiron_base/pull/1326 and https://github.com/pyiron/pyiron_base/pull/1325 - these contain the changes to the database queries but not the refactoring done in this pull request.